src/types.cpp

```diff
@@ -830,4 +831,32 @@ namespace librealsense
     {
         snapshot = std::make_shared<color_sensor_snapshot>();
     }
+
+    std::string frame_to_string(const frame_interface * f)
+    {
+        std::ostringstream s;
+        auto composite = dynamic_cast<const composite_frame *>(f);
+        if (composite)
+        {
+            for (int i = 0; i < composite->get_embedded_frames_count(); i++)
+            {
+                s << "[" <<
```

Comment:
`[` should be before the loop, ']' after.
No space please.
Do you really need to do frame_to_string? Don't you have operator<<()?